elements/rh-card/demo/inline-promo.html

```diff
@@ -0,0 +1,85 @@
+<rh-card variant="inline-promo">
+  <svg slot="inline-promo" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 190">
+    <title>A placeholder image in a card header slot</title>
+    <rect x="0" y="0" width="320" height="190" fill="#e7f0f9"/>
+    <line x1="0" y1="0" x2="320" y2="190" stroke="#d6e7f5"/>
+    <line x1="320" y1="0" x2="0" y2="190" stroke="#d6e7f5"/>
+  </svg>
+
+  <h2>Featured: left aligned with image</h2>
+  <p>You can customize the <code>#body</code> by using the <code>::part</code> selector. Using this technique, you can have padding on the top, bottom, left, right, or none at all.</p>
+  <rh-cta>
```

Comment:
```suggestion
  <rh-cta slot="footer">
```

Reply:
Adding `slot="footer"` to the CTA breaks the design/layout. Can you explain this suggestion further?

Reply:
my thought process is that
- ctas usually  go in card footers
- lots of existing html has cta in card footer
- so toggling the variant to inline-promo should (maybe) work?

Reply:
Indeed, that is a logical thought process and seems like it could certainly happen in the wild.

Technically, you _could_ put the CTA in the `footer` slot and style it ala:

```css
rh-card[variant="inline-promo"]::part(footer) {
    grid-column: 2;
}
```

Then it looks like this:

![Inline promo with CTA in footer - spacing visualized](https://github.com/RedHat-UX/red-hat-design-system/assets/575865/60443135-c639-4b17-a9bc-8de91b34beeb)

Here's what's happening with the grid behind the scenes:

![Inline promo with CTA in footer - CSS grid visualized](https://github.com/RedHat-UX/red-hat-design-system/assets/575865/0c70bb24-30e8-4e75-9b2e-b377dd29f08b)

Unfortunately, it's [not possible](https://stackoverflow.com/questions/45264354/is-it-possible-to-place-more-than-one-element-into-a-css-grid-cell-without-overl) to place more than one grid element into a cell without overlap. CSS grid doesn't have a concept of flow.

Hopefully, when people see the CTA in the wrong spot, they'll remove `slot="footer"` from the CTA (fixing their layout), actualize peak web component mastery and move on to sweating each pixel in their image for the Inline Promo `image` slot. 

Let me know how you would like to proceed.

Reply:
I ran some experiments and it seems like the way forward is:

1. when the variant is inline-promo, the footer slot moves to becomes a child of the body container and loses it's margins.
2. the difference between a large inline promo and a large full-width promo layout is only a matter of container query.. i.e. a "full width" large inline promo is one that's larger than a given width.
3. a "standard" inline promo is one that has a body and a footer but no image or header.

Going this way, I believe it's possible that the only things users will have to change to get these patterns is the content, and in the case of the "standard" pattern, the color-palette attr, and even that we could enforce by requiring lighter bg color even for lightest when the content conditions for standard obtain.

Reply:
@coreyvickery @marionnegp note that according to the approach i lay out above, users will have to apply page css in order to get the full width (i.e. borderless) pattern on a some screen sizes _in the case where the promo is full width within it's container, but not full page width_ i.e. there's a sidebar. Is that intended behavior?

Reply:
consensus from office hours:
- it's `promo` not `inline-promo`
- `promo` is indeed a variant of card
- when promo doesn't have a heading or image, it appears like the "standard" type in the figma
- we'll use the approach above to align the content model of promo to card